spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -148,6 +159,23 @@ public boolean isRunning() {
 		return new KafkaProducer<K, V>(this.configs, this.keySerializer, this.valueSerializer);
 	}
 
+	@Override
+	public KafkaSender<K, V> createReactiveSender() {
+		if (this.sender == null) {
+			synchronized (this) {
+				if (this.sender == null) {
+					this.sender = createSender();
+				}
+			}
+		}
+		return this.sender;
+	}
+
+	protected CloseSafeKafkaSender<K, V> createSender() {
+		SenderOptions<K, V> options = SenderOptions.create(this.configs);
+		return new CloseSafeKafkaSender<>(reactor.kafka.sender.internals.ProducerFactory.INSTANCE, options);
```

Comment:
It would be fairly hacky, but wondering if we can't simply pass a `ProducerFactory` that returns `this.producer` (and maybe revise Reactor Kafka to allow a constructor that takes a Producer as argument). It seems like this creates two separate producers managed by this class (case in which this also opens a discussion about user expectations wrt ordering if mixing reactive/non-reactive sends, flush, etc) - perhaps it would be easier to have a separate reactive template after all.